Feature/road shape

RichardKarlFuchs opened the pull request:
Bisher noch ohne Mesh

RichardKarlFuchs committed:
add first method to sample evenly spaced points along a list of beziercurves

RichardKarlFuchs committed:
add bezier scene, places spheres along a bezier curve

RichardKarlFuchs committed:
add calculation of length

RichardKarlFuchs committed:
optimize memory usage

RichardKarlFuchs committed:
remove unnecessary increment of length

RichardKarlFuchs committed:
set Lane_Width to be 3 meters

RichardKarlFuchs committed:
specify road points by position and forward Vector

RichardKarlFuchs committed:
Merge branch 'develop' into feature/road_shape

RichardKarlFuchs committed:
Normalize first forward vector of points Array, suggested by @sidlertime

RichardKarlFuchs committed:
add comments to methods

RichardKarlFuchs committed:
move calculation of length insinde of calculation of evenly spaced points

RichardKarlFuchs committed:
add last point to array of evenly spaced points

RichardKarlFuchs committed:
set length to match number of points in array

RichardKarlFuchs committed:
Merge branch 'develop' into feature/road_shape

RichardKarlFuchs committed:
set length calculation to be exact, add endingPoint (very last Point, not necessary at an even distance)

RichardKarlFuchs committed:
add global illumination to bezier scene

RichardKarlFuchs committed:
add roadShape that takes all parameters and creates copy

RichardKarlFuchs committed:
add Revert Method to Bezier Curve

RichardKarlFuchs committed:
add method to invert a roadShape

RichardKarlFuchs committed:
change method to invert BezierCurves

RichardKarlFuchs committed:
rename Edge.position to Edge.originPoint

RichardKarlFuchs committed:
alter length to reference RoadShape.length

RichardKarlFuchs committed:
alter AbsolutePosition to use RoadShape

RichardKarlFuchs committed:
add roadShape construction in Simulation Controller

RichardKarlFuchs committed:
alter roadSpawner to use originPoint.position instead of position

RichardKarlFuchs committed:
remove angle from Edge

RichardKarlFuchs committed:
remove creation of road with positions 1 and 2

RichardKarlFuchs committed:
set Angle when moving and spawning cars based on forward property of roadPoint

RichardKarlFuchs committed:
Clamp positionOnRoad to avoid IndexOutOfBoundsExeption

RichardKarlFuchs committed:
Make positions of Beziercurve in Bezier-Scene public

RichardKarlFuchs committed:
fix calculation of interpolated points on roadShape, adjust bezier resolution to finer interval

RichardKarlFuchs committed:
Add Inversion of forward vectors

RichardKarlFuchs committed:
fix bug in calculation linear interpolation for non integer positionOnRoad

RichardKarlFuchs committed:
rename perpandicularOffset to perpendicularOffset

RichardKarlFuchs committed:
add control over controlPoint in Editor

RichardKarlFuchs committed:
adjust roadSpawner to include curvature of road

Sidlertime reviewed (approved):
Vielleicht in einer invertierungssache: könnte sein dass da immernoch 
foreach curve in reversedcurves
Steht, müsste aber in curves sein

RichardKarlFuchs committed:
fix bug in creation of inverted list of beziercurves

rrickfox commented:
> 
> 
> Vielleicht in einer invertierungssache: könnte sein dass da immernoch
> foreach curve in reversedcurves
> Steht, müsste aber in curves sein

Erledigt, ce9f029

funketh commented on Assets/Scripts/DataTypes/Edges/BezierCurve.cs:
```diff
@@ -0,0 +1,47 @@
+            }
+
+            return calculatedPoints;
+        }
```
was ist der sinn von `lastPoint`?

funketh commented on Assets/Scripts/DataTypes/Edges/RoadShape.cs:
```diff
@@ -1,7 +1,100 @@
-        
+        private List<BezierCurve> _curves;
+        public RoadPoint[] points;
+        public RoadPoint endingPoint;
```
welchen Sinn hat `endingPoint`? (wird nie genutzt)

funketh committed:
general cleanup

- use var for temporary variables
- remove redundancies
- use LINQ to shorten collection construction

funketh committed:
shorten BezierCurve.Revert()

funketh reviewed (dismissed):
Der Code sollte allgemein kommentiert werden

RichardKarlFuchs committed:
Merge branch 'develop' into feature/road_shape

RichardKarlFuchs committed:
add missing call to Utility namespace

RichardKarlFuchs committed:
remove lastPoint

RichardKarlFuchs committed:
add comments

RichardKarlFuchs committed:
add usage of endingPoint, used when positionOnRoad superseeds the last evenly spaced point

RichardKarlFuchs committed:
add 2 curves to pointPlacer

funketh committed:
simplify BezierCurve.CalculatePoints() using the yield keyword

funketh commented:
klär mal die merge-konflikte ab

funketh commented:
Die performance der simulation ist bei mir durch die kurve immens gesunken.
Wäre es nicht sinnvoller, ein einziges rundes 3D Objekt zu nutzen, anstatt manuell sehr viele Würfel zu spawnen? (Ist das das was du mit Mesh meinst? Kenne mich da nicht aus)

rrickfox commented:
Performance sollte durch das verwenden von einem Mesh wieder steigen. Im Moment werden halt die 7000 Würfel, die eine Straße darstellen, alle vollständig und einzeln gerendert, anstatt dass es nur ein Straßenobjekt gibt.

RichardKarlFuchs committed:
simplify roadShape by snapping to fixed length

RichardKarlFuchs committed:
fix spelling mistakes

RichardKarlFuchs committed:
Merge branch 'develop' into feature/road_shape

# Conflicts:
#	Assets/Scripts/DataTypes/Car.cs
#	Assets/Scripts/DataTypes/Edges/Edge.cs
#	Assets/Scripts/DataTypes/Vertices/EndPoint.cs
#	Assets/Scripts/DataTypes/Vertices/Vertex.cs
#	Assets/Scripts/RoadSpawner.cs
#	Assets/Scripts/SimulationController.cs
#	Assets/Scripts/Utility/CONSTANTS.cs

RichardKarlFuchs committed:
fix spelling mistake

RichardKarlFuchs committed:
fix display errors

RichardKarlFuchs committed:
remove unnecessary parameters in GetAbsolutePosition
